### PR TITLE
ascanrules: Format String scan rule refactor

### PR DESCRIPTION
**Pull request by kingthorin**
- FormatStringScanRule > Added individual alert creation methods so that the ref only needs to be maintained in a single location. Also removed one unnecessary jump.

Follow-up to: https://github.com/zaproxy/zap-extensions/pull/4623

### CODE REVIEW

**Comment by thc202** [comment]
The refactor looks good but this is all the same alert, I don't think references apply here.

**Comment by kingthorin** [comment]
So keep the refactoring but don't bother with setting refs?

**Comment by kingthorin** [comment]
Tweaked

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/FormatStringScanRule.java [hunk 1]
```diff
@@ -286,22 +285,34 @@ public int getWascId() {
         return 6;
     }
 
-    private AlertBuilder createAlert(
-            String attack, String param, HttpMessage msg, Character errorChar) {
+    private AlertBuilder createBaseAlert(
+            String attack, String param, HttpMessage msg, Character ref) {
```
Comment (ricekot) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/FormatStringScanRule.java:289]: Revert parameter name to `errorChar`?

Comment (kingthorin) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/FormatStringScanRule.java:289]: Sure, I’ll tweak that in a few hours.

**Comment by kingthorin** [comment]
Tweaked

**Review by ricekot — APPROVED** [review]
Thank you!

**Pull request merged**